Sources/DistributedSystem/ChannelCompression.swift

```diff
@@ -357,47 +321,52 @@ final class ChannelCompressionHandshakeClient: ChannelInboundHandler, RemovableC
     }
 
     func channelActive(context: ChannelHandlerContext) {
-        self.logger.debug("\(context.channel.addressDescription)/\(Self.self): channel active")
+        logger.debug("\(context.channel.addressDescription)/\(Self.self): channel active")
         let compressionMode = distributedSystem.compressionMode
+        let promise = context.eventLoop.makePromise(of: Void.self)
         switch compressionMode {
         case .disabled:
             var buffer = ByteBufferAllocator().buffer(capacity: MemoryLayout<HandshakeRequest.RawValue>.size)
             buffer.writeInteger(HandshakeRequest.noCompression.rawValue)
-            context.writeAndFlush(NIOAny(buffer), promise: nil)
+            context.writeAndFlush(NIOAny(buffer), promise: promise)
```

Comment:
```swift
let buffer = switch compressionMode {
...
}

let promise = context.eventLoop.makePromise(of: Void.self)
context.writeAndFlush(NIOAny(buffer), promise: promise)
```
?

Probably can even extract last 2 lines to a separate extension for `ChannelHandlerContext`:
```swift
extension ChannelHandlerContext {
    func writeAndFlush(_ buffer: ByteBuffer) -> Promise<Void> {
        let promise = self.eventLoop.makePromise(of: Void.self)
        self.writeAndFlush(NIOAny(buffer), promise: promise)
        return promise
    }
}
```
And this can be used in other places too, e.g. `ChannelCompressionHandshakeServer.sendResponse()`

Reply:
Well, the problem is that we not only send different data depending on the compression mode, but also modify pipeline in a different way.